### PR TITLE
Add Appian Compare Service to SEATool compare

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose

Following the patterns in this project for the compare service for MMDL vs SEATool. We created the compare-appian service that compares Appian and SEATool data. Business logic was implemented within the steps.

Business logic:
https://qmacbis.atlassian.net/browse/OY2-20888

#### Linked Issues to Close
https://qmacbis.atlassian.net/browse/OY2-22901

## Approach
- Used the patterns established by Ben in the Compare service. Renamed our service to compare-appian. Can work on changing any other names as we see fit.

### CODE REVIEW

File: src/services/compare-appian/handlers/getAppianData.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    data.appianRecord = appianRecord;
+    data.SPA_ID = appianRecord.payload?.SPA_ID;
+
+    /* Checking if the appian record was signed within the last 200 days. */
+    const submissionDate = appianRecord.payload?.SBMSSN_DATE;
+    data.secSinceAppianSubmitted = secondsBetweenDates(submissionDate);
+
+    // data.programType = programType;
+    data.appianSubmitted =
+      appianRecord.payload?.IS_SBMTD?.toLowerCase() === "y";
```
Comment (benjaminpaige) [inline src/services/compare-appian/handlers/getAppianData.ts:24]: lol. funny

File: src/services/compare-appian/handlers/sendDoesNotExistAlert.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  console.log("Received event:", JSON.stringify(event, null, 2));
+
+  const region = process.env.region;
+  const project = process.env.project;
+  const stage = process.env.stage;
+
+  if (!region) throw "process.env.region needs to be defined.";
+
+  // use this secret path to define the { emailRecipients, sourceEmail } for the does not exist email
+  const secretId = `${project}/${stage}/alerts`;
```
Comment (benjaminpaige) [inline src/services/compare-appian/handlers/sendDoesNotExistAlert.ts:25]: we may want to make these secrets appian specific. i know wale has adjusted the name for mmdl to be mmdl-alerts so maybe something like `const secretId = `${project}/${stage}/appian-alerts` then we dont have to worry about classes across features.

File: src/services/compare-appian/handlers/sendNoMatchAlert.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  console.log("Received event:", JSON.stringify(event, null, 2));
+
+  const region = process.env.region;
+  const project = process.env.project;
+  const stage = process.env.stage;
+
+  if (!region) throw "process.env.region needs to be defined.";
+
+  // use this secret path to define the { emailRecipients, sourceEmail } for the does not match email
+  const secretId = `${project}/${stage}/alerts`;
```
Comment (benjaminpaige) [inline src/services/compare-appian/handlers/sendNoMatchAlert.ts:25]: same as above

File: src/services/compare-appian/handlers/sendReport.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  SPA_ID: string;
+  iterations: number;
+  secSinceAppianSubmitted: string;
+  appianSubmittedDate: string;
+  seatoolExist: boolean;
+  seatoolSubmissionDate?: string;
+  match?: boolean;
+}
+
+//! This work/logic will be done in another ticket
```
Comment (benjaminpaige) [inline src/services/compare-appian/handlers/sendReport.ts:20]: perfect. yeah i thin k we want to have a separate status table for this comparison. we should also make the one that exists mmdl specific in its naming conventions. We can do that together, it shouldnt take more than an hour.

File: src/services/compare-appian/serverless.yml [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+  commands:
+    stop-executions: |
+      aws stepfunctions list-executions \
+      --state-machine-arn arn:aws:states:${self:provider.region}:${aws:accountId}:stateMachine:${self:service}-${sls:stage}-compare-alerting \
+      --status-filter RUNNING \
+      --query "executions[*].{executionArn:executionArn}" \
+      --output text | \
+      xargs -I {} aws stepfunctions stop-execution \
+      --execution-arn {} >> out.t
+      rm out.t
```
Comment (benjaminpaige) [inline src/services/compare-appian/serverless.yml:110]: i found out this script has issues, so we opted for a different pattern. we are using a custom resource instead, if you follow this change: https://github.com/CMSgov/seatool-compare/pull/88/files i think the only thing that would be different is the `stateMachineArn: !Ref MMDLAlertingStateMachine`

**Review by benjaminpaige — COMMENTED** [review]
this looks great. I just have a few suggestions for updates. Also lets plan an hour or so this week to see if we can get together toreconsile some of the naming conventions. Things should either be appian or mmdl (i.e status tables, compare service, putLogEvent type, etc)

File: src/services/compare-appian/handlers/getAppianData.ts (hunk 1 above)
Comment (jdinh8124) [reply src/services/compare-appian/handlers/getAppianData.ts:23]: You're telling me haha

**Review by benjaminpaige — APPROVED** [review]
Looks great! don't forget to use the semantic commit language and SQUASH!

**Comment by benjaminpaige** [comment]
Oh @jdinh8124 I just noticed... you prob want to add an `appian-status` table before you merge and use that instead of the `status` table, they will be identical except for name. lmk if you have questions.

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**